### ISSUE TITLE
cannot install dependencies

### ISSUE THREAD

**Issue by prankousky** [athityakumar/colorls#234]
Hi,

I am running `colorls` on macOS Mojave Version 10.14. `ruby --version` is `ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]`, `rbenv --version` is `rbenv 1.1.1`.

Whenever I run colorls, I get this output

```
Ignoring clocale-0.0.3 because its extensions are not built. Try: gem pristine clocale --version 0.0.3
Ignoring eventmachine-1.2.7 because its extensions are not built. Try: gem pristine eventmachine --version 1.2.7
Ignoring ffi-1.9.25 because its extensions are not built. Try: gem pristine ffi --version 1.9.25
Ignoring http_parser.rb-0.6.0 because its extensions are not built. Try: gem pristine http_parser.rb --version 0.6.0
Ignoring unf_ext-0.0.7.5 because its extensions are not built. Try: gem pristine unf_ext --version 0.0.7.5
```

I tried running those commands as stated, also tried `gem pristine --all` (was recommended to me when asking in ruby subreddit on reddit), also tried running all those commands as root instead of regular user.

No success.

Any ideas on how to get rid of these errors when trying to run colorls?

Thanks in advance for your help 👍

**Comment by avdv** [comment]
Hi.

According to https://apple.stackexchange.com/a/167377

> This problem occurs, if a gem was compiled against a different libruby as the currently running.

Did you install the gems system wide? You should install them into the rvm managed gems directory.

Try to remove all those gems listed and re-install colorls. Make sure to use the same Ruby version when installing and running.

**Comment by prankousky** [comment]
Thank you.

I installed all gems systemwide, or at least I think so... I need to be root to install, otherwise I get `ERROR:  While executing gem ... (Gem::FilePermissionError) 
  You don't have write permissions for the /Library/Ruby/Gems/2.3.0 directory.`

I tried everything mentioned in that link, but no success. How can I make sure to use the same Ruby version when installing and running? If I go by `which ruby` and use that exact file (in this case `/usr/bin/ruby`), I get the same errors as if I were to just run `ruby` without an exact path.

Sorry, I am absolutely new to ruby and (unless there are some gems I run without noticing) exclusively installed it to use colorls. So I have no knowledge on how to do anything else with it. I installed it following the docs, so I cannot pinpoint the part where things went wrong.

**Comment by prankousky** [comment]
I solved it. Not sure what actually did the trick, but this is what I did:

1. remove rbenv (`brew uninstall rbenv`)
2. try installing colorls for user only (`gem install --user-install colorls`): got an error about insufficient rights. apparently, colorls was installed in my user directory **with root permissions**, so it could not be updated
3. deleted entire colorls directory from my local gem folder
4. re-installed with the user-install command

still not success...

5. i did not have `~/.gem/ruby/2.3.0/bin` in my PATH. i don't know how and why it had disappeared as it did work before without any trouble. so i added it to my environment variables and now colorls works fine again.

thank you :)
